client/components/proposal/AddActionFunctionForm.tsx

```diff
@@ -90,6 +97,27 @@ export const AddActionFunctionForm = ({
 
   return (
     <form onSubmit={submitHandler}>
+      { isProxy && (
+        <>
+        { implementationAddress ? (
+          <div className="form-control w-full mt-2">
+            <span className="text-xs m-1 font-semibold">Contract selected is a proxy</span>
+            <label className="label bg-gray-100 p-3 rounded-lg cursor-pointer" htmlFor="implementation-methods">
+              <span className="label-text">Select from implementation methods</span>
```

Comment:
Lets make this a dropdown (similar to the one on the first page) called `Implementation contract` that preselects the implementation that is detected.

Our vault uses a double proxy setup so the proxy implementation will be detected as the Vault contract, but that doesn't include a lot of callable methods from VaultCore. So I think we should allow them to select any contract for the implementation to handle that (unless you have a nicer way).

Reply:
Great. Yea the current vault setup doesn't implement `implementation` for both, and I haven't seen any other contracts built in the same way, so it is in itself an edge case unfortunately.